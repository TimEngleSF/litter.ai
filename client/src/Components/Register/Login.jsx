import React, { useState, useContext } from 'react';
import { Link } from 'react-router-dom';
import UserContext from '../../utils/UserContext';
import { fetchLogin, fetchLogOut } from '../../utils/fetchUserData';

export default function Login({ setUser, user }) {
  const userContext = useContext(UserContext);
  console.log('userContext: ', userContext);
  const [loginData, setLoginData] = useState({
    email: '',
    password: '',
  });

  const handleLogin = async () => {
    try {
      await userContext.login(loginData);
    } catch (err) {
      console.error(err);
    }
    // try {
    //   const userData = await fetchLogin(loginData);
    //   await setUser(userData);
    // } catch (err) {
    //   console.error(err);
    // }
  };

  const handleChange = (e) => {
    e.preventDefault();
    setLoginData({ ...loginData, [e.target.id]: e.target.value });
  };

  const handleCheck = () => {
    console.log('user: ', user);
  };

  return (
    <div>
      <form>
        <label htmlFor="email">
          email:
          <input
            type="text"
            id="email"
            value={loginData.email}
            onChange={(e) => handleChange(e)}
          />
        </label>
        <label htmlFor="password">
          password:
          <input
            type="text"
            id="password"
            value={loginData.password}
            onChange={(e) => handleChange(e)}
          />
        </label>
        <button type="button" onClick={handleLogin}>Login</button>
        <button type="button" onClick={handleCheck}>Test Login Data</button>
        <button type="button" onClick={() => fetchLogOut()}>Log out</button>
        <Link to="/leaderboard"><button type="button">Leaderboard</button></Link>
      </form>
    </div>
  );
}
