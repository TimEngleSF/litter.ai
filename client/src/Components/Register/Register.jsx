import React from 'react';

export default function Register({ setUser }) {
  const handleRegisterClick = (e) => {
    e.preventDefault();
    const link = 'http://localhost:3001/login';

    const fetchData = async () => {
      try {
        const res = await fetch(link);
        const response = await res.json();
        setUser(response);
      } catch (err) {
        console.error(err);
      }
    };
    fetchData();
  };

  return (
    <div>
      <button type="button" onClick={handleRegisterClick}>Register</button>
    </div>
  );
}
