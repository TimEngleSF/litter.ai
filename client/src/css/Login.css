.login-container {
  width: 100%;
}

.login-logo {
  display: flex;
  width: 100%;
  height: 30dvh;
  background-color: var(--main-green);
  margin-bottom: 1.5rem;
  align-items: center;
  justify-content: center;
}

.login-logo-image{
  max-height: 10rem;
  grid-column-start: 2;
  grid-row-start: 2;
  object-fit: contain;
}

.login-form {

}

.invalid-login {
  color: rgb(255, 0, 0);
  text-align: center;
}

.login-email, .login-password {
  width: 50%;;
  margin: 10px auto;
}

.login-email label, .login-password label {
  color: gray;
  font-size: small;
}

.login-email input, .login-password input {
  display: block;
  width: 100%;
}