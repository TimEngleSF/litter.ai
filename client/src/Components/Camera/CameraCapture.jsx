import React, { useState } from 'react';
import ImageUploaded from './ImageUploaded';
import AddImage from './AddImage';
import '../../css/Capture.css';

export default function CameraCapture() {
  const [image, setImage] = useState(null);

  return (
    <div className="capture-container main-container">
      {image
        ? <ImageUploaded image={image} setImage={setImage} />
        : <AddImage setImage={setImage} />}
      <script src="ImageCapture.js" />
    </div>
  );
}
