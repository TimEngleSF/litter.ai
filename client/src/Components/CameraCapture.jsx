import React, { useState, useRef } from 'react';
import { Link } from 'react-router-dom';
import { CiCirclePlus } from 'react-icons/ci';
import Dropdown from './Dropdown';
import '../css/Capture.css';

export default function CameraCapture() {
  const [image, setImage] = useState(null);
  const [imageSubmitted, setImageSubmitted] = useState(false);
  const [imageCategory, setImageCategory] = useState('glass');
  const captureRef = useRef(null);

  const openCamera = () => {
    captureRef.current.click();
  };

  const removePhoto = () => {
    setImage(null);
  };

  const handleImage = (e) => {
    const imageData = [e.target.files[0]];
    setImage({
      imagePreview: URL.createObjectURL(new Blob(imageData, { type: 'application/zip' })),
      imageFile: e.target.files[0],
    });
  };

  const handlePhotoSubmit = () => {
    setImageSubmitted(true);
  };

  return (
    <div className="capture-container main-container">
      {/* Image has been uploaded; preview the image */}
      {image
        ? (
          <div className="capture-image-wrapper">
            <div className="capture-image-preview">
              <button
                type="button"
                className="image-x-button"
                onClick={removePhoto}
              >
                &#x2715;
              </button>
              <img alt="" src={image.imagePreview} className="capture-image" />
            </div>
            {/* Image has been uploaded and submitted to the AI */}
            {imageSubmitted
              ? (
                <div className="capture-classify">
                  <h1>Help us classify</h1>
                  <div className="capture-dropdown">
                    <Dropdown
                      setImageCategory={setImageCategory}
                      imageCategory={imageCategory}
                    />
                  </div>
                  <p>
                    Is this&nbsp;
                    {imageCategory}
                    ?
                    If it is not,
                    please select the correct category from the dropdown.
                  </p>
                  <div className="capture-confirm-button lower-buttons">
                    <Link to={`/success/${imageCategory}`}>
                      <button type="button">
                        Confirm
                      </button>
                    </Link>
                  </div>
                </div>
              )
              : (
                <div className="capture-submit-button lower-buttons">
                  <button
                    type="button"
                    onClick={handlePhotoSubmit}
                  >
                    Submit
                  </button>
                </div>
              )}
          </div>
        )
        /* Image has not been uploaded, opens device's camera or gives option to upload an image */
        : (
          <>
            <div className="capture-top-section">
              <CiCirclePlus onClick={openCamera} className="capture-icon" />
              <input
                type="file"
                accept="image/*"
                id="cameraFileInput"
                ref={captureRef}
                capture="environment"
                onChange={handleImage}
                style={{ display: 'none' }}
              />
              <div className="capture-icon-bg" />
              <h1>Take a photo of the item</h1>
              <p>
                <span>Click on the camera to open up your device&#x27;s camera.</span>
                <br />
                <span>Please take the photo in portrait mode for best results.</span>
              </p>
            </div>
            <div className="capture-buttons lower-buttons">
              <Link to="/">
                <button className="button-home" type="button">
                  Home
                </button>
              </Link>
            </div>
          </>
        )}
      <script src="ImageCapture.js" />
    </div>
  );
}
