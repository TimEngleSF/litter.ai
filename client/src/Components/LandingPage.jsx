import React from 'react';
import { PiFlowerTulipThin } from 'react-icons/pi';
import { Link } from 'react-router-dom';
import '../css/Landing.css';

export default function LandingPage({ user }) {
  return (
    <>
      <div className="landing-container main-container">
        <div className="landing-top-section">
          <PiFlowerTulipThin className="landing-logo" />
          <div className="landing-text">
            <h1>Welcome to LitterSort</h1>
            <p>With just a photo, sort your garbage and save the world.</p>
          </div>
          <div className="landing-top-buttons lower-buttons">
            <Link to="/capture">
              <button type="button" className="landing-capture-button">
                Capture Picture
              </button>
            </Link>
          </div>
        </div>
        <div className="landing-bottom-section">
          <div className="landing-bottom-buttons lower-buttons">
            <Link to="/leaderboard">
              <button type="button" className="landing-secondary-button">
                Leaderboard
              </button>
            </Link>
            { user
              ? (
                <Link to="/profile">
                  <button type="button" className="landing-secondary-button">
                    Profile
                  </button>
                </Link>
              )
              : (
                <>
                  <Link to="/register">
                    <button type="button" className="landing-secondary-button">
                      Sign Up
                    </button>
                  </Link>
                  <div className="landing-login-text">
                    <p>Already have an account?</p>
                    <Link to="/login">
                      Log in
                    </Link>
                  </div>
                </>
              )}
          </div>
        </div>
      </div>
      <div className="landing-full-bg" />
    </>
  );
}
