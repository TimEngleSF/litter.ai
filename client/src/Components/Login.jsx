import React, { useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { FaAngleLeft, FaRegEnvelope } from 'react-icons/fa6';
import { FiLock } from 'react-icons/fi';
import { fetchLogin } from '../utils/fetchUserData';
import '../css/LoginSignUp.css';

export default function Login({ setUser }) {
  const [loginData, setLoginData] = useState({
    email: '',
    password: '',
  });

  const [invalidLogin, setInvalidLogin] = useState(false);
  const navigate = useNavigate();

  const handleLogin = async () => {
    try {
      const response = await fetchLogin(loginData);
      setUser(response.user.displayUsername);
      setInvalidLogin(false);
      navigate('/');
    } catch (err) {
      setInvalidLogin(true);
      console.error(err);
    }
  };

  const handleChange = (e) => {
    e.preventDefault();
    setLoginData({ ...loginData, [e.target.id]: e.target.value });
  };

  const goBack = () => {
    navigate(-1);
  };

  return (
    <div className="login-signup-container main-container">
      <div className="login-signup-header">
        <button className="back-button" type="button" aria-label="Back" onClick={goBack}>
          <FaAngleLeft />
        </button>
        <div className="login-signup-header-text">
          <h3>Sign In</h3>
          <p>Enter your credentials to continue</p>
        </div>
      </div>

      <form className="login-signup-form">
        {invalidLogin
          ? (
            <div className="invalid-login-signup">
              <small>Incorrect username and/or password</small>
            </div>
          )
          : <br className="invalid-login-signup" />}
        <div className="login-signup-email">
          <FaRegEnvelope className="login-signup-icon" />
          <input
            type="email"
            id="email"
            aria-label="Email"
            placeholder="Email"
            value={loginData.email}
            onChange={(e) => handleChange(e)}
          />
        </div>
        <br className="login-signup-form-break" />
        <div className="login-signup-password">
          <FiLock className="login-signup-icon" />
          <input
            type="password"
            id="password"
            aria-label="Password"
            placeholder="Password"
            minLength="4"
            value={loginData.password}
            onChange={(e) => handleChange(e)}
          />
        </div>
      </form>

      <div className="lower-buttons login-signup-buttons">
        <button
          type="button"
          onClick={handleLogin}
          id="login-signup-button"
        >
          Login
        </button>
        <div className="sign-up">
          <small>Don&apos;t have an account?</small>
          <Link
            to="/register"
            className="register-button"
          >
            <small>Sign Up</small>
          </Link>
        </div>
      </div>
    </div>
  );
}
