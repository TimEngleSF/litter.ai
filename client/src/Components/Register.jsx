import React, { useState } from 'react';
import { GrLocation } from 'react-icons/gr';
import { FaRegUser } from 'react-icons/fa';
import { FaAngleLeft, FaRegEnvelope } from 'react-icons/fa6';
import { FiLock } from 'react-icons/fi';
import { PiUserListBold } from 'react-icons/pi';
import { Link, useNavigate } from 'react-router-dom';
import { fetchRegister } from '../utils/fetchUserData';
import '../css/LoginSignUp.css';

export default function Register({ setUser }) {
  const [registerData, setRegisterData] = useState({
    firstName: '',
    lastName: '',
    email: '',
    username: '',
    password: '',
    confirmPassword: '',
    zipCode: '',
  });
  const [validPassword, setValidPassword] = useState(false);
  const [passwordMatch, setPasswordMatch] = useState(true);

  const navigate = useNavigate();

  const handleRegister = async () => {
    if (passwordMatch && validPassword) {
      try {
        const response = await fetchRegister(registerData);
        setUser(response.user.displayUsername);
        navigate('/');
      } catch (err) {
        console.error(err);
      }
    }
  };

  const handleChange = (e) => {
    e.preventDefault();
    setRegisterData({ ...registerData, [e.target.id]: e.target.value });
  };

  const handlePasswordCheck = () => {
    const [pass, confPass] = [registerData.password, registerData.confirmPassword];

    // If passwords don't match, user will be notified.
    if (confPass.length && pass !== confPass) {
      setPasswordMatch(false);
    } else if (pass === confPass) {
      setPasswordMatch(true);
    }

    // If password is shorter than 6 characters, user will be notified.
    if (pass.length && pass.length >= 6) {
      setValidPassword(true);
    } else if (pass.length && pass.length < 6) {
      setValidPassword(false);
    }
  };

  const goBack = () => {
    navigate(-1);
  };

  return (
    <div className="login-signup-container main-container">
      <div className="login-signup-header">
        <button className="back-button" type="button" aria-label="Back" onClick={goBack}>
          <FaAngleLeft />
        </button>
        <div className="login-signup-header-text">
          <h3>Create a New Account</h3>
          <p>Join us in making a cleaner world</p>
        </div>
      </div>

      <form className="login-signup-form">
        <div className="login-signup-first-name">
          <PiUserListBold className="login-signup-icon" />
          <input
            type="text"
            id="firstName"
            aria-label="First Name"
            placeholder="First Name"
            value={registerData.firstName}
            onChange={(e) => handleChange(e)}
            required
          />
        </div>
        <br className="login-signup-form-break" />
        <div className="login-signup-last-name">
          <PiUserListBold className="login-signup-icon" />
          <input
            type="text"
            id="lastName"
            aria-label="Last Name"
            placeholder="Last Name"
            value={registerData.lastName}
            onChange={(e) => handleChange(e)}
            required
          />
        </div>
        <br className="login-signup-form-break" />
        <div className="login-signup-email">
          <FaRegEnvelope className="login-signup-icon" />
          <input
            type="email"
            id="email"
            aria-label="Email"
            placeholder="Email"
            value={registerData.email}
            onChange={(e) => handleChange(e)}
            required
          />
        </div>
        <br className="login-signup-form-break" />
        <div className="login-signup-zipCode">
          <GrLocation className="login-signup-icon" />
          <input
            type="zipCode"
            id="zipCode"
            aria-label="Zip Code"
            placeholder="Zip Code"
            value={registerData.zipCode}
            onChange={(e) => handleChange(e)}
            required
          />
        </div>
        <br className="login-signup-form-break" />
        <div className="login-signup-username">
          <FaRegUser className="login-signup-icon" />
          <input
            type="text"
            id="username"
            aria-label="Username"
            placeholder="Username"
            value={registerData.username}
            onChange={(e) => handleChange(e)}
            required
          />
        </div>
        <br className="login-signup-form-break" />
        <div className="login-signup-password">
          <FiLock className="login-signup-icon" />
          <input
            type="password"
            id="password"
            aria-label="Password"
            placeholder="Password"
            minLength="6"
            value={registerData.password}
            onChange={(e) => handleChange(e)}
            onKeyUp={handlePasswordCheck}
            required
          />
        </div>
        {(registerData.password.length > 0 && !validPassword)
          && (
            <div className="valid-password">
              <small>Password must be a minimum of 6 characters</small>
            </div>
          )}
        <br className="login-signup-form-break" />
        <div className="login-signup-confirm-password">
          <FiLock className="login-signup-icon" />
          <input
            type="password"
            id="confirmPassword"
            aria-label="Confirm Password"
            placeholder="Confirm Password"
            value={registerData.confirmPassword}
            onChange={(e) => handleChange(e)}
            onKeyUp={handlePasswordCheck}
            required
          />
        </div>
        {!passwordMatch
          && (
            <div className="matching-password">
              <small>Passwords do not match</small>
            </div>
          )}
      </form>

      <div className="lower-buttons login-signup-buttons">
        <button
          type="button"
          onClick={handleRegister}
          id="login-signup-button"
        >
          Register
        </button>
        <Link to="/">
          <button className="button-home" type="button">
            Home
          </button>
        </Link>
      </div>
    </div>
  );
}
