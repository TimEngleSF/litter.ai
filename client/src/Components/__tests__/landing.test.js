/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  screen,
  fireEvent,
} from '@testing-library/react';
import { MemoryRouter as Router, Routes, Route } from 'react-router-dom';
import LandingPage from '../LandingPage';
import CameraCapture from '../Camera/CameraCapture';
import Register from '../Register';
import Login from '../Login';

describe('Home component', () => {
  const landingPage = <Router><LandingPage /></Router>;
  const user = 'Briana30';

  test('Landing page matches the current snapshot', () => {
    const tree = renderer.create(landingPage);
    expect(tree).toMatchSnapshot();
  });

  test('Logo and introduction text loads on the screen', () => {
    render(landingPage);
    expect(screen.getByTestId('litterai-logo')).toBeInTheDocument();
    expect(screen.getByText('Welcome to LitterSort')).toBeInTheDocument();
    expect(screen.getByText('With just a photo, sort your garbage and save the world.')).toBeInTheDocument();
  });

  test("Clicking on 'Capture Picture' navigates to capture page", async () => {
    render(
      <Router>
        <Routes>
          <Route path="/" element={<LandingPage user={user} />} />
          <Route path="/capture" element={<CameraCapture />} />
        </Routes>
      </Router>,
    );

    expect(screen.getByTestId('litterai-logo')).toBeInTheDocument();
    fireEvent.click(screen.getByRole('button', { name: 'Capture Picture' }));
    expect(screen.getByText('Take a photo of the item')).toBeInTheDocument();
  });

  test('Clicking on Sign Up button navigates to the registration page', async () => {
    render(
      <Router>
        <Routes>
          <Route path="/" element={<LandingPage />} />
          <Route path="/register" element={<Register />} />
        </Routes>
      </Router>,
    );

    expect(screen.getByTestId('litterai-logo')).toBeInTheDocument();
    await fireEvent.click(screen.getByRole('button', { name: 'Sign Up' }));
    expect(screen.getByRole('button', { name: /register/i })).toBeInTheDocument();
  });

  test('Clicking on Log in button navigates to the Login page', async () => {
    render(
      <Router>
        <Routes>
          <Route path="/" element={<LandingPage />} />
          <Route path="/login" element={<Login />} />
        </Routes>
      </Router>,
    );

    expect(screen.getByText('Already have an account?')).toBeInTheDocument();
    fireEvent.click(screen.getByRole('link', { name: 'Log in' }));
    expect(screen.getByRole('button', { name: 'Login' })).toBeInTheDocument();
  });
});
