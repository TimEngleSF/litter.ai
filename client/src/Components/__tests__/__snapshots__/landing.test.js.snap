// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Landing page matches the current snapshot 1`] = `
Array [
  <div
    className="landing-container main-container"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="landing-top-section"
    >
      <img
        alt="litterai-logo-white"
        className="landing-logo"
        data-testid="litterai-logo"
        onLoad={[Function]}
        src="Logo_large_white.png"
      />
      <div
        className="landing-text"
      >
        <h1>
          Welcome to LitterSort
        </h1>
        <p>
          With just a photo, sort your garbage and save the world.
        </p>
      </div>
      <div
        className="landing-buttons lower-buttons"
      >
        <a
          href="/capture"
          onClick={[Function]}
        >
          <button
            className="landing-capture-button"
            type="button"
          >
            Capture Picture
          </button>
        </a>
      </div>
    </div>
    <div
      className="landing-bottom-section"
    >
      <div
        className="landing-buttons lower-buttons"
      >
        <p
          className="landing-signup-description"
        >
          Create an account to track your statistics and compete against others!
        </p>
        <a
          href="/register"
          onClick={[Function]}
        >
          <button
            className="landing-secondary-button"
            type="button"
          >
            Sign Up
          </button>
        </a>
        <div
          className="landing-login-text"
        >
          <p>
            Already have an account?
          </p>
          <a
            href="/login"
            onClick={[Function]}
          >
            Log in
          </a>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="landing-full-bg"
  />,
]
`;
